src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java

```diff
@@ -1,274 +1,150 @@
     }
 
     public Multi<EventType> getEventTypesByEndpointId(@NotNull String accountId, @NotNull UUID endpointId) {
-        String query = "SELECT et.id AS et_id, et.name AS et_name, et.display_name AS et_din, et.description AS et_description, a.id AS a_id, a.name AS a_name, a.display_name as a_displayName, a.bundle_id as a_bundle_id FROM public.event_type et " +
-                "JOIN public.applications a ON a.id = et.application_id " +
-                "JOIN public.endpoint_targets endt ON  endt.event_type_id = et.id " +
-                "WHERE endt.endpoint_id = $1 AND endt.account_id = $2";
-
-        return connectionPublisher.get().onItem()
-                .transformToMulti(c -> Multi.createFrom().resource(() -> c,
-                        c2 -> {
-                            Flux<PostgresqlResult> execute = c2.createStatement(query)
-                                    .bind("$1", endpointId)
-                                    .bind("$2", accountId)
-                                    .execute();
-
-                            return mapResultSetToEventTypes(execute);
-                        })
-                        .withFinalizer(postgresqlConnection -> {
-                            postgresqlConnection.close().subscribe();
-                        }));
+        String query = "SELECT e FROM EventType e LEFT JOIN FETCH e.application JOIN e.targets t " +
+                "WHERE t.id.accountId = :accountId AND t.endpoint.id = :endpointId";
+        return session.createQuery(query, EventType.class)
+                .setParameter("accountId", accountId)
+                .setParameter("endpointId", endpointId)
+                .getResultList()
+                .onItem().transformToMulti(Multi.createFrom()::iterable);
     }
 
-    private Flux<EventType> mapResultSetToEventTypes(Flux<PostgresqlResult> resultFlux) {
-        return resultFlux.flatMap(postgresqlResult -> postgresqlResult.map((row, rowMetadata) -> {
-            Application app = new Application();
-            app.setBundleId(row.get("a_bundle_id", UUID.class));
-            app.setId(row.get("a_id", UUID.class));
-            app.setName(row.get("a_name", String.class));
-            app.setDisplay_name(row.get("a_displayName", String.class));
-
-            EventType eventType = new EventType();
-            eventType.setId(row.get("et_id", UUID.class));
-            eventType.setName(row.get("et_name", String.class));
-            eventType.setDisplay_name(row.get("et_din", String.class));
-            eventType.setDescription(row.get("et_description", String.class));
-
-            eventType.setApplication(app);
-
-            return eventType;
-        }));
+    /**
+     * Adds to the given {@link Application} a reference to a persistent {@link Bundle} without actually loading its
+     * state from the database. The app will remain unchanged if it does not contain a non-null bundle identifier.
+     *
```

Comment:
Like above: does bundle==null make sense here?

Reply:
Same kind of answer: it does not seem to be an existing use-case from what I just saw in the code.

These methods were originally introduced because Hibernate Reactive failed in quite an ugly and non-helping way when `session.getReference(SomeEntity.class, null)` was called. I can't remember the exact exception though.